framework_tvb/tvb/adapters/uploaders/tract_importer.py

```diff
@@ -265,15 +271,15 @@ def launch(self, view_model):
                 if not tractf.endswith('.txt'):  # omit directories and other non track files
                     continue
                 vertices_file = zipf.open(tractf)
-                tract_vertices = numpy.loadtxt(vertices_file, dtype=numpy.float32)
+                datatype.tract_vertices = numpy.loadtxt(vertices_file, dtype=numpy.float32)
 
-                tract_start_indices.append(tract_start_indices[-1] + len(tract_vertices))
-                datatype.store_data_chunk("vertices", tract_vertices, grow_dimension=0, close_file=False)
+                tract_start_indices.append(tract_start_indices[-1] + len(datatype.tract_vertices))
+                tracts_h5.write_vertices_slice(datatype.tract_vertices)
 
-                if region_volume_index is not None:
-                    tract_region.append(self._get_tract_region(tract_vertices[0]))
+                if view_model.region_volume is not None:
+                    tract_region.append(self._get_tract_region(datatype.tract_vertices[0]))
                 vertices_file.close()
 
         datatype.tract_start_idx = tract_start_indices
-        datatype.tract_region = numpy.array(tract_region, dtype=numpy.int16)
+        datatype.tract_region = numpy.array(tract_region)
```

Comment:
Why not keep int16 here? I believe the region numbers will fit inside int16

Reply:
I thought we discussed to not explicitly put the size and let numpy choose the smallest size that it can fit the data into.

Reply:
That was available only for `tract_start_idx` where values won't fit into int16. But for `tract_region` we know values would fit and numpy probably won't use dtype=int16 by default.